Core/include/Acts/EventData/detail/Charge.hpp

```diff
@@ -115,41 +137,91 @@ struct SinglyCharged {
   }
 };
 
+/// Charge and momentum interpretation for arbitrarily charged but not neutral
+/// particles.
+class AnyButNoCharge {
+ public:
+  /// Construct with the magnitude of the input charge.
+  constexpr AnyButNoCharge(float absQ) noexcept : m_absQ{absQ} {
+    assert((0 < absQ) and "Input charge magnitude must be positive");
+  }
+  constexpr AnyButNoCharge(SinglyCharged) noexcept : m_absQ{UnitConstants::e} {}
+
+  constexpr float absQ() const noexcept { return m_absQ; }
+
+  template <typename T>
+  constexpr auto qFromQOP(T qOverP) const noexcept {
+    // using because of autodiff
+    using std::copysign;
+    return copysign(m_absQ, qOverP);
+  }
+  template <typename T>
+  constexpr auto pFromQOP(T qOverP) const noexcept {
+    // using because of autodiff
+    using std::abs;
+    return qFromQOP(qOverP) / qOverP;
+  }
+
+  template <typename P, typename Q>
+  constexpr auto qopFromPQ(P p, Q q) const noexcept {
```

Comment:
I didn't want to make an assumption about the sign of the particle. But we have a case here and there where we want to get the `qop` from a known `q` and `p` which is easy but also a bit cumbersome with the special case for neutral particles.

I think the only use case of having this as a member is that you will not violate the charge hypothesis using this interface. Otherwise this could be a single global function I guess